### PR TITLE
Create redux action for search

### PR DESCRIPTION
**Pull request by rottencandy**
Adds:
- Redux actions and reducers for updating store with searched name
- Connect `Homepage` component to dispatch action on form submit.

### CODE REVIEW

File: src/Homepage.tsx [hunk 1]
```diff
@@ -21,12 +23,22 @@ const appLogo = {
   target: '_blank'
 }
 
-const Homepage: React.FC = () => (
+const mapDispatchToProps = (dispatch: Dispatch) => ({
```
Comment (invincibleJai) [inline src/Homepage.tsx]: will be good to have type here

File: src/Homepage.tsx [hunk 2]
```diff
@@ -21,12 +23,22 @@ const appLogo = {
...
   <Page header={<Header logo={appLogo} />}>
     <PageSection>
-      <SearchBar onFormSubmit={() => {}} />
+      <SearchBar onFormSubmit={fetchDetails} />
     </PageSection>
   </Page>
 )
 
-export default Homepage
+export default connect(null, mapDispatchToProps)(Homepage)
```
Comment (invincibleJai) [inline src/Homepage.tsx]: will be good to have type here

File: src/redux/reducers/name.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { SearchNameType, SEARCH_NAME } from '../types'
+
+export function nameReducer (state = '', action: SearchNameType) {
```
Comment (invincibleJai) [inline src/redux/reducers/name.ts]: what's the use of this reducer?

Comment (rottencandy) [reply src/redux/reducers/name.ts]: Was planning to turn this into thunk that would make API call.

File: src/Homepage.tsx (hunk 1 above)
Comment (rottencandy) [reply src/Homepage.tsx]: Added

File: src/Homepage.tsx (hunk 2 above)
Comment (rottencandy) [reply src/Homepage.tsx]: Updated

File: src/Homepage.test.tsx [hunk 4]
```diff
@@ -1,11 +1,17 @@
...
+  const enzymeWrapper = mount(<Homepage />)
+  return { enzymeWrapper }
+}
 
-it('renders without crashing', () => {
-  mount(<Homepage />)
+it('should render itself and subcomponents', () => {
+  const { enzymeWrapper } = setup()
+  expect(enzymeWrapper.exists('Page')).toBe(true)
+  expect(enzymeWrapper.exists('PageSection')).toBe(true)
```
Comment (invincibleJai) [inline src/Homepage.test.tsx]: can check `toHaveLength`

Comment (rottencandy) [reply src/Homepage.test.tsx]: Updated

**Pull request merged**